fiftyone/core/odm/document.py

```diff
@@ -649,63 +657,112 @@ def _save(
 
         ensure_connection()
 
-        if validate:
-            self.validate(clean=clean)
-
         id_field = self._meta["id_field"]
-
-        created = self._created
-        if not created:
-            created = "_id" not in self.to_mongo(fields=[id_field])
-
-        doc = self.to_mongo()
-        _id = doc.get("_id", None)
+        _id = self.to_mongo(fields=[id_field]).get("_id", None)
+        created = self._created or _id is None
 
         ops = None
 
         if created:
-            # Save new document
+            # Insert new document
+            if validate:
+                self._validate(clean=clean)
+
+            doc = self.to_mongo()
+
             if _id is None:
                 _id = ObjectId()
                 doc["_id"] = _id
 
-            if deferred:
-                ops = [InsertOne(doc)]
-            else:
-                self._get_collection().insert_one(doc)
+            self[id_field] = self._fields[id_field].to_python(_id)
+            self._created = False
+
+            ops = self._insert(doc, deferred=deferred)
         else:
             # Update existing document
             updates = {}
-            sets, unsets = self._delta()
 
-            if sets:
-                updates["$set"] = sets
+            try:
+                # OPTIMIZATION: we monkey patch `to_mongo()` and
```

Comment:
Apologies for this, I couldn't help myself...

This significantly improves the efficiency of `sample.save()` if a sample contains many large fields but you've only made a simple edit to one field.